classes/Search.php

```diff
@@ -0,0 +1,572 @@
+
+        if (!defined('FRONT_END_SEARCH') || !FRONT_END_SEARCH) {
+            return array(
+                'error' =>'Apologies, search has been turned off currently for performance reasons.'
+            );
+        }
+
+        if (!$SEARCHENGINE->valid) {
+            return array('error' => $SEARCHENGINE->error);
+        } else {
+            $LIST = new \HANSARDLIST();
+            $data = $LIST->display('search', $args , 'none');
+            $data['sort_order'] = $sort_order;
+            $data['members'] = $members;
+            $data['cons'] = $cons;
+            $data['glossary'] = $glossary;
+            return $data;
+        }
+    }
+
+    private function search_order_p($searchstring) {
+        global $DATA, $this_page;
+
+        $q_house = '';
+        if (ctype_digit(get_http_var('house'))) {
+            $q_house = get_http_var('house');
+        }
+
+        $wtt = get_http_var('wtt');
+        if ($wtt) {
+            $q_house = 2;
+        }
+
+        # Fetch the results
+        $data = search_by_usage($searchstring, $q_house);
+
+        if ($wtt) {
+            $q_house = 2;
+            $pagetitle = 'League table of Lords who say ' . $data['pagetitle'];
+        } else {
+            $pagetitle = 'Who says ' . $data['pagetitle'] . ' the most?';
+        }
+        $DATA->set_page_metadata($this_page, 'title', $pagetitle);
+
+
+        $data['house'] = $q_house;
+        $data['wtt'] = $wtt;
+        return $data;
+    }
+
+    private function find_constituency($args) {
+        if ($args['s'] != '') {
+            $searchterm = $args['s'];
+        } else {
+            return false;
+        }
+
+        list ($constituencies, ) = search_constituencies_by_query($searchterm);
+
+        $constituency = "";
+        if (count($constituencies)==1) {
+            $constituency = $constituencies[0];
+        }
+
+        $cons = array();
+        try {
+            if ($constituency != '') {
+                // Got a match, display....
+
+                $MEMBER = new Member(array('constituency'=>$constituency, 'house' => 1));
+                $cons[] = $MEMBER;
+            } elseif (count($constituencies)) {
+                foreach ($constituencies as $constituency) {
+                    $MEMBER = new Member(array('constituency'=>$constituency, 'house' => 1));
+                    $cons[] = $MEMBER;
+                }
+            }
+        } catch ( MemberException $e ) {
+            $cons = array();
+        }
+
+        return $cons;
+    }
+
+    private function find_members($searchstring) {
+        $searchstring = trim(preg_replace('#-?[a-z]+:[a-z0-9]+#', '', $searchstring));
+        $q = search_member_db_lookup($searchstring);
+        if (!$q) return array();
+
+        $members = array();
+        if ($q->rows() > 0) {
+            $row_count = $q->rows();
+            for ($n=0; $n<$row_count; $n++) {
+                $member = new Member(array('person_id' => $q->field($n, 'person_id')));
```

Comment:
Okay, so this is certainly much nicer code than previous :) So I see where code of my first comment is coming from. But then the second query (after the name search) of search_member_db_lookup isn't doing anything here, so perhaps that should be factored into two functions, maybe along with something to fetch multiple Members at once or similar?